### PR TITLE
Feature: Switch blocks improvement / Uncles improvement

### PR DESCRIPTION
**Pull request by olgakup**

### CODE REVIEW

File: apps/ethvm/src/components/blocks/largeBlocks/blockDetail.vue [hunk 1]
```diff
@@ -119,7 +140,7 @@ export default Vue.extend({
         {
           title: this.$i18n.t('block.miner'),
           detail: this.block.getMiner(),
-          link: '/address/' + this.block.getMiner(),
+          link: '/address/0x' + this.block.getMiner(),
```
[the remark this answers is not in the document]
Comment (aldoborrero) [reply apps/ethvm/src/components/blocks/largeBlocks/blockDetail.vue]: In general in this file there's harcoded 0x, so move those accordingly to the model

**Comment by aldoborrero** [comment]
@niktrix and @olgakup 

Some comments:
- The message when switching between blocks should be:
    - If we're on the latest block and we navigate forward into the future (block is being mined): " Progress spinner - This Block has not been mined yet..."
    - If we're loading already mined blocks and we move backwards and forward. The message should be: "Progress spinner - Loading Block..."
- Arrows for navigating forward and backwards are disabled outside block detail page (if we're in uncle page or pending txs it doesn't make sense to have those enabled) 
- There are in block detail some information that is not being displayed correctly but the db has the info already.
- For uncle detail we should remove Uncles > button and change titles to Uncle Detail. Also the the table, the alternating colors are not being correctly done from row Parent Hash (it follows: white | grey | white | white). Look the screenshot below.
- I propose to change date format what etherscan has
- We should add Uncle Height and Uncle Position information in Uncle detail
- Let's keep consistent the display of ETH (let's use 3 digits for both Miner Reward and Uncle reward)
- If the block transactions are loading we should display a progress that says: "Progress spinner - Loading transactions...". If there are none: "This block does not have any transactions" or the list of transactions
- Fix for block detail the error: Uncle(s) SHA3 0xundefined

screen shot 2018-12-03 at 13 48 49

**Comment by aldoborrero** [comment]
@olgakup I remember you mentioned past week about the breadcrumb you wanted to work on that as well. Currently in Uncles the breadcrumb is not displaying correctly the message, should @niktrix fix it? Or should leave as is until you review the feature?

File: apps/ethvm/src/components/blocks/largeBlocks/blockDetail.vue [hunk 2]
```diff
@@ -84,8 +84,8 @@
     </div>
     <v-card flat v-else class="pa-3">
       <v-layout column align-center justify-center ma-3>
-        <v-card-title class="primary--text text-xs-center body-2 pb-4"> Sit tight, this Block has not been mined yet </v-card-title>
         <v-icon class="fa fa-spinner fa-pulse fa-4x fa-fw primary--text" large></v-icon>
+        <v-card-title v-if="isNotMinedBlock" class="primary--text text-xs-center body-2 pb-4"> Sit tight, this Block has not been mined yet </v-card-title>
```
Comment (aldoborrero) [inline apps/ethvm/src/components/blocks/largeBlocks/blockDetail.vue]: Add this message using vue.i18n

File: apps/ethvm/src/components/blocks/largeBlocks/blockDetail.vue [hunk 3]
```diff
@@ -252,7 +261,10 @@ export default Vue.extend({
   },
   computed: {
     isUncle() {
-      return this.block.getIsUncle()
+      if(this.block){
```
Comment (aldoborrero) [inline apps/ethvm/src/components/blocks/largeBlocks/blockDetail.vue]: return this.Block && this.block.getIsUncle()

File: apps/ethvm/src/components/blocks/largeBlocks/blockDetail.vue [hunk 4]
```diff
@@ -281,8 +281,8 @@ export default Vue.extend({
       return this.isMined
     },
     formatTime(){
-      let date = new Date(this.block.getTimestamp())
-      return date
+      let date = new Date(this.block.getTimestamp()).toString()
```
Comment (aldoborrero) [inline apps/ethvm/src/components/blocks/largeBlocks/blockDetail.vue]: @niktrix prefer const

File: apps/ethvm/src/components/blocks/largeBlocks/blockDetail.vue [hunk 5]
```diff
@@ -22,7 +22,7 @@
                   <v-layout row justify-start align-center fill-height>
                     <v-card-title class="info--text pr-0 pl-0">{{ $t('common.hash') }}:</v-card-title>
                     <v-card-text class="text-truncate">
-                      <router-link :to="'/block/' + uncles[index]"> 0x{{ uncles[index].unclesHash }} </router-link>
+                      <router-link :to="'/uncle/0x' + uncles[index]"> 0x{{ uncles[index] }} </router-link>
```
Comment (aldoborrero) [inline apps/ethvm/src/components/blocks/largeBlocks/blockDetail.vue]: @niktrix Remove hardcoded 0x and embed that inside the model

Comment (niktrix) [reply apps/ethvm/src/components/blocks/largeBlocks/blockDetail.vue]: fixed

File: apps/ethvm/src/components/blocks/largeBlocks/latestBlocks.vue [hunk 6]
```diff
@@ -37,91 +40,100 @@
...
+                  </p>
+                </v-flex>
+              </v-layout>
+              <v-layout row v-if="hasUncles(block)" pl-3 pr-2 pt-0 pb-1>
+                <v-flex d-flex hidden-xs-only sm2 pt-0 pr-0>
+                  <v-img v-if="hasUncles(block)" :src="require('@/assets/uncle.png')" height="30px" contain></v-img>
+                </v-flex>
+                <v-flex xs12 sm7 md6>
+                  <v-card flat color="uncleGrey">
+                    <v-card-title class="pt-1 font-weight-medium">Uncles:</v-card-title>
```
Comment (aldoborrero) [inline apps/ethvm/src/components/blocks/largeBlocks/latestBlocks.vue]: Change this use v18n

File: apps/ethvm/src/translations/en_US.json [hunk 7]
```diff
@@ -99,7 +102,9 @@
     "recRoot": "Receipts Root",
     "uncReward": "Reward",
     "sha": "SHA3",
-    "uncle": "Uncle(s)"
+    "uncle": "Uncle(s)",
+    "loadingBlockTx":"Loading Block transactions...",
+    "notMined":"Sit tight, this Block has not been mined yet"
```
Comment (aldoborrero) [inline apps/ethvm/src/translations/en_US.json]: Change it: This block has not been mined yet

**Comment by aldoborrero** [comment]
@niktrix several more comments:

- [ ]  We should fix State Root (now it displays [object ArrayBuffer] and its entry is duplicated twice if you're on uncle detail page.
- [ ] Is components/blocks/largeBlocks/uncleDetail.vue being used? If not, delete it

**Pull request merged**